ui-tests/src/test/java/io/syndesis/qe/steps/integrations/IntegrationSteps.java

```diff
@@ -0,0 +1,366 @@
+
+	@Then("^Camilla is presented with \"(\\w+)\" integration details$")
+	public void verifyIntegrationDetails(String integrationName) {
+		//WHERE IS IntegrationDetailPage ???
+		assert integrationName.equals(detailPage.getIntegrationName()) : "Integration detail editPage must show integration name";
+		// todo add more assertion on integration details editPage
+	}
+
+	@When("^Camilla selects the \"(\\w+)\" integration.*$")
+	public void selectConnection(String itegrationName) {
+		listPage.listComponent().goToIntegrationDetail(itegrationName);
+	}
+
+	@When("^she selects \"(\\w+)\" integration action$")
+	public void selectIntegrationAction(String action) {
+		if ("Create Opportunity".equals(action)) {
+			log.warn("Action {} is not available", action);
+			listActions.selectAction("Create Salesforce object");
+		}
+		listActions.selectAction(action);
+	}
+
+	@When("^Camilla deletes the \"(\\w+)\" integration*$")
+	public void deleteIntegration(String integrationName) {
+		listComponent.clickDeleteIntegration(integrationName);
+	}
+
+	@When("^Camilla deletes the integration on detail page*$")
+	public void deleteIntegrationOnDetailPage() {
+		detailPage.deleteIntegration();
+	}
+
+	@Then("^she can see on detail editPage that integration is \"(\\w+)\" status$")
+	public void checkStatus(String expectedStatus) {
+		String status = detailPage.getStatus();
+		log.info("Status: {}", status);
+		assert expectedStatus.equals(status) : "Status on integration detail page should be equal to expected status";
+	}
+
+	@Then("^she clicks on integration in \"(\\w+)\" status and check on detail if status match and appropriate actions are available$")
+	public void clickOnIntegrationInStatus(String status) {
+		SelenideElement integrationByStatus = listComponent.getIntegrationByStatus(status);
+		integrationByStatus.shouldBe(visible).click();
+
+		//WAITING FOR IntegrationDetailPageFactory()
+		const integrationDetailPageFactory = new IntegrationDetailPageFactory();
+		IntegrationDetailPage detailPage2 = integrationDetailPageFactory.getDetailPage(status);
+
+		for (String action : detailPage2.actionsSet) {
+			log.info("Action: {}", action);
+			SelenideElement actionButton = detailPage2.getActionButton(action);
+			log.info("There should by button for {} action on {} status", action, status);
+			actionButton.shouldBe(visible);
+
+			log.info("Status on detail editPage should be equal to expected status");
+			assert status.equals(detailPage2.getStatus()) : "Status on detail editPage should be equal to expected status";
+		}
+		detailPage2.getActionButton("Done").shouldBe(visible).click();
+	}
+
+	@Then("^she go trough whole list of integrations and check on detail if status match and appropriate actions are available$")
+	public void goTrouhListAndCheckDetails() {
+		ElementsCollection integrations = listComponent.getAllIntegrations();
+
+		for (SelenideElement integration : integrations) {
+//			const integration = listComponent.getAllIntegrations().get(i);
```

Comment:
There're code snippets left-overs through the whole file. Do you want to keep them for reference?

Reply:
they are to be deleted, before merge.